test/unittest/middleware/ced/CedMiddlewareTests.cpp

```diff
@@ -0,0 +1,907 @@
+    std::string xml_two{"xml_two"};
+
+    /* Create Participant by REF. */
+    EXPECT_TRUE(middleware_.create_participant_by_xml(0, 0, xml_one));
+
+    /* Create:
+     *      Id:         same
+     *      Domain:     same
+     *      XML:        same
+     *      Expected:   FALSE
+     */
+    EXPECT_FALSE(middleware_.create_participant_by_xml(0, 0, xml_one));
+
+    /* Create:
+     *      Id:         different
+     *      Domain:     same
+     *      XML:        same
+     *      Expected:   TRUE
+     */
+    EXPECT_TRUE(middleware_.create_participant_by_xml(1, 0, xml_one));
+
+    /* Create:
+     *      Id:         same
+     *      Domain:     different
+     *      XML:        same
+     *      Expected:   FALSE
+     */
+    EXPECT_FALSE(middleware_.create_participant_by_xml(0, 1, xml_one));
+
+    /* Create:
+     *      Id:         same
+     *      Domain:     same
+     *      XML:        different
+     *      Expected:   FALSE
+     */
+    EXPECT_FALSE(middleware_.create_participant_by_xml(0, 0, xml_two));
+}
+
+TEST_F(CedMiddlewareUnitTests, DeleteParticipant)
+{
+    std::string ref{"Participant"};
+    middleware_.create_participant_by_ref(0, 0, ref);
+
+    /* Delete
+     *      Participant:    existent
+     *      Expected:       TRUE
+     */
+    EXPECT_TRUE(middleware_.delete_participant(0));
+
+    /* Delete
+     *      Participant:    non-existent
+     *      Expected:       FALSE
+     */
+    EXPECT_FALSE(middleware_.delete_participant(0));
+
+    /* Create
+     *      Participant:    same
+     *      Expected:       TRUE
+     */
+    EXPECT_TRUE(middleware_.create_participant_by_ref(0, 0, ref));
+}
+
+TEST_F(CedMiddlewareUnitTests, CreateTopicByRef)
+{
+    std::string participant_ref{"Participant"};
+    middleware_.create_participant_by_ref(0, 0, participant_ref);
+    middleware_.create_participant_by_ref(1, 0, participant_ref);
+
+    std::string topic_ref_one{"TopicOne"};
+    std::string topic_ref_two{"TopicTwo"};
+
+    /* Create Topic by REF. */
+    EXPECT_TRUE(middleware_.create_topic_by_ref(0, 0, topic_ref_one));
+
+    /* Create:
+     *      Id:             same
+     *      Participant:    same
+     *      Reference:      same
+     *      Expected:       FALSE
+     */
+    EXPECT_FALSE(middleware_.create_topic_by_ref(0, 0, topic_ref_one));
+
+    /* Create:
+     *      Id:             different
+     *      Participant:    same
+     *      Reference:      same
+     *      Expected:       FALSE
+     */
+    EXPECT_FALSE(middleware_.create_topic_by_ref(1, 0, topic_ref_one));
+
+    /* Create:
+     *      Id:             same
+     *      Participant:    different
+     *      Reference:      same
+     *      Expected:       TRUE
+     */
+    EXPECT_TRUE(middleware_.create_topic_by_ref(0, 1, topic_ref_one));
+
+    /* Create:
+     *      Id:             same
+     *      Participant:    same
+     *      Reference:      different
+     *      Expected:       TRUE
```

Comment:
And what is the behaviour? Is it overwritten?

Reply:
Yes, it was the expected behaviour but to avoid confusion a change it, and the new behaviour is analogous to the publisher and subscriber objects.